### ISSUE TITLE
IsExpression mapping for parent type property does not work

### ISSUE THREAD

**Issue by arwyl** [linq2db/linq2db#1543]
If there are parent entity and child entity, when I set mapping for child entity and map some parent entity field to an expression, it cannot be resolved

```
public class SomeEntityBase
{
	public int Id { get; set; }
	public int SomeIntField { get; set; }
}

public class SomeChildEntity : SomeEntityBase
{
	public int SomeOtherField { get; set; }
}

[Test]
public void DatasourcesTestsPerf111()
{
	var builder = new MappingSchema().GetFluentMappingBuilder();

	var emb = builder.Entity<SomeChildEntity>();

	emb.HasTableName($"SomeTable");

	emb.Property(x => x.Id).IsColumn().HasColumnName("SomeId");
	// some expression
	emb.Property(x => x.SomeIntField).IsExpression(x => x.Id + 123);

	var schema = builder.MappingSchema;

	using (var db = new DataConnection(GetDataProvider(), "some connection string", schema))
	{
		var qq = db.GetTable<SomeChildEntity>()
			.Where(x => x.SomeIntField > 200)
			.Select(x => new
			{
				x.Id,
				x.SomeIntField
			})
			.Take(1)
			;
		qq.ToList();
	}
}
```

Exception:
```
Message: LinqToDB.Linq.LinqException : Expression 'x.SomeIntField' (x.SomeIntField) is not a table.
```

If I change 
var emb = builder.Entity<SomeChildEntity>();
to 
var emb = builder.Entity<SomeEntityBase>();
everything works, but it seems strange and I also cannot cache mapping scheme if I have multiple child entities with different parent property mappings.

**Comment by arwyl** [comment]
@sdanyliv also checked it with your fix for #1523 - the bug remains

**Comment by MaceWindu** [comment]
Fixed by #1544
